From 6637996d330a821847f83907db9dd751983deff9 Mon Sep 17 00:00:00 2001
From: Levi Zim <rsworktech@outlook.com>
Date: Wed, 28 May 2025 20:47:02 +0800
Subject: [PATCH 1/2] cpuinfo: enable for riscv64 linux

In crrev.com/c/5256319 cpuinfo was disabled for riscv64 due to lack of
support. Now cpuinfo supports riscv64 linux and we can enable it.
---
 third_party/cpuinfo/BUILD.gn    | 12 ++++++++++++
 third_party/cpuinfo/cpuinfo.gni |  1 -
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/third_party/cpuinfo/BUILD.gn b/third_party/cpuinfo/BUILD.gn
index 7fc0f7324ebfd..6b759d57f718d 100644
--- a/third_party/cpuinfo/BUILD.gn
+++ b/third_party/cpuinfo/BUILD.gn
@@ -108,6 +108,10 @@ source_set("cpu_specific") {
     ]
   }
 
+  if (current_cpu == "riscv64") {
+    sources = [ "src/src/riscv/uarch.c" ]
+  }
+
   # Because "init.c" is reused again, but only has sources in the corresponding
   # build config.
   deps = [ ":x86_cache" ]
@@ -175,6 +179,14 @@ source_set("cpu_and_os_specific") {
     ]
   }
 
+  if ((is_linux || is_chromeos) && current_cpu == "riscv64") {
+    sources = [
+      "src/src/riscv/linux/init.c",
+      "src/src/riscv/linux/riscv-hw.c",
+      "src/src/riscv/linux/riscv-isa.c",
+    ]
+  }
+
   if (is_android && (current_cpu == "arm" || current_cpu == "arm64")) {
     sources = [
       "src/src/arm/android/properties.c",
diff --git a/third_party/cpuinfo/cpuinfo.gni b/third_party/cpuinfo/cpuinfo.gni
index f7a27ee82601b..be4e1554f540b 100644
--- a/third_party/cpuinfo/cpuinfo.gni
+++ b/third_party/cpuinfo/cpuinfo.gni
@@ -6,7 +6,6 @@ use_cpuinfo =
     # cpuinfo is not supported on these chips.
     current_cpu != "ppc64" && current_cpu != "ppc64le" &&
     current_cpu != "s390" && current_cpu != "s390x" &&
-    current_cpu != "riscv64" &&
     # cpuinfo is not supported on fuchsia.
     !is_fuchsia &&
     # There's a few arm architectures that are not supported by cpuinfo,
-- 
2.39.2

